src/ripple/app/misc/AmendmentTableImpl.cpp

```diff
@@ -151,28 +236,27 @@ AmendmentTableImpl::get (std::string const& name)
     return uint256 ();
 }
 
-AmendmentState*
-AmendmentTableImpl::addKnown (const char* amendmentID, const char* friendlyName,
-    bool veto)
+bool
+AmendmentTableImpl::addKnown (uint256 const& hash,
+                              std::string const& friendlyName,
+                              bool veto)
```

Comment:
I'm with Tom on this one (although I haven't sold Vinnie yet).  I've seen cases in rippled code where four bools are being passed through call chains.  It's frightfully easy to mis-order those bools as they go from place to place.  If you make a (very light weight) type (e.g., and enum), then the compiler can tell you whether you've messed up the order.  and you get better naming for free.

If you follow Tom's advice, you might make it an `enum class Veto : unsigned char ...`, since the underlying type of a scoped enum is `int` unless you declare otherwise.